classes/task/populate_objects_filesize.php

```diff
@@ -0,0 +1,52 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_objectfs\task;
+
+use core\task\adhoc_task;
+
+/**
+ * Ad-hoc task to update objects table. Used for async upgrade.
+ *
+ * @package    tool_objectfs
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2022 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class populate_objects_filesize extends adhoc_task {
+
+    /**
+     * Action of task.
+     */
+    public function execute() {
```

Comment:
This one worries me as I'd expect it to take forever on a big site.

There is two ways it could be better:
1) rewrite it as pure sql so we don't need to pull data down and push it back

The problem with this is that it's not cross db compliant, BUT we can write two code paths so it does the more performant sql in places where we know it works

```
INSERT INTO {foo} (clientid, domain, usermodified, timecreated, timemodified)
                SELECT id as clientid, domain, usermodified, timecreated, timemodified
                  FROM {bar} client
                 WHERE NOT EXISTS (
                     SELECT 1
                       FROM {foo}
                      WHERE domain = client.domain)"
```

there is precedent for this in core:
https://github.com/moodle/moodle/blob/master/mod/quiz/locallib.php#L1148-L1179

2) The other way is we just page the data and limit it to N updates and then spawn another adhoc task if there are more to do